sway-lib-std/src/message.sw

```diff
@@ -0,0 +1,49 @@
+library message;
+
+use ::alloc::alloc;
+use ::outputs::{Output, output_count, output_type};
+use ::mem::{addr_of, copy};
+use ::revert::revert;
+use ::vec::Vec;
+use ::vm::evm::evm_address::EvmAddress;
+
+const FAILED_SEND_MESSAGE_SIGNAL = 0xffff_ffff_ffff_0002;
+
+/// Sends a message to `recipient` of length `msg_len` through `output` with amount of `coins`
+///
+/// # Arguments
+///
+/// * `recipient` - The address of the message recipient
+/// * `msg_data` - arbitrary length message data
+/// * `coins` - Amount of base asset sent
+pub fn send_message(recipient: EvmAddress, msg_data: Vec<u64>, coins: u64) {
```

Comment:
@mohammadfawaz `smo` takes a `recipient` param which is a `b256`, so a `b256` param would work just as well in the `send_message` wrapper function.
`smo` will create a receipt, which can then be parsed to extract the relevent data as part of the bridging process. So the `recipient` is always going to be an Evm Address as the point of `smo` is to allow bridging between Fuel and some other layer. Using an `EvmAddress` here adds some type-safety, ensuring that developers set the recipient parm correctly.
Now that I think about it, if we ever want to bridge to another L1, this may need to be rethought? 🤔 
A simple `b256` is more flexible.

Reply:
thoughts on this @pixelcircuits ? (recipient param type, `b256` or `EvmAddress`?)

Reply:
Ok, reverting back to a `b256` here as I think it's better than enshrining the EVM in a general-purpose stdlib function.